internal/gatewayapi/filters.go

```diff
@@ -152,9 +153,18 @@ func (t *Translator) processURLRewriteFilter(
 	filterContext *HTTPFiltersContext,
 ) {
 	if filterContext.URLRewrite != nil {
-		if filterContext.URLRewrite.Hostname != nil ||
-			filterContext.URLRewrite.Path.FullReplace != nil ||
-			filterContext.URLRewrite.Path.PrefixMatchReplace != nil {
+		// avoid conflicts:
+		// 1. multiple core URLRewriteFilters (of any type)
+		// 2. HTTPRouteFilter and URLRewriteFilter must not change the same component (host, path)
+
+		hasCoreRewrites := filterContext.URLRewrite.Path != nil && (filterContext.URLRewrite.Path.FullReplace != nil ||
+			filterContext.URLRewrite.Path.PrefixMatchReplace != nil) || (filterContext.URLRewrite.Host != nil && filterContext.URLRewrite.Host.Name != nil)
+		hasExtensionPathRewrites := filterContext.URLRewrite.Path != nil && filterContext.URLRewrite.Path.RegexMatchReplace != nil
+		hasExtensionHostRewrites := filterContext.URLRewrite.Host != nil && (filterContext.URLRewrite.Host.Header != nil ||
+			filterContext.URLRewrite.Host.Backend != nil)
+
+		if (rewrite.Hostname != nil && (hasCoreRewrites || hasExtensionHostRewrites)) ||
+			(rewrite.Path != nil && (hasCoreRewrites || hasExtensionPathRewrites)) {
```

Comment:
any chace to improve these lines, it's hard to read/understand( `conditionA && (conditionB || conditionC)`).